### ISSUE TITLE
TB78 compatibility

### ISSUE THREAD

**Issue by brenthuisman** [autocrypt-thunderbird/autocrypt-thunderbird#21]
With the release of TB78, this plugin does not work anymore. Is any maintenance planned at this point?

**Comment by je-vv** [comment]
So, no plans to make it work on TB v78 and beyond?  I used Enigmail with had enough autocrypt functionality, but now that TB v78 no longer supports it, neither autocrypt-thunderbird.  It's really sad cause of the decision TB made of not supporting it.  From [TB-PGP-support](https://support.mozilla.org/en-US/kb/openpgp-thunderbird-howto-and-faq#w_does-thunderbird-support-autocrypt):

> Thunderbird does not support the Autocrypt philosophy that encryption should be fully automatic

The only thing TB will keep supporting is header messages and importing keys...  So it doesn't seem it'll come to TB, and perhaps an extension is needed, not to encrypt itself, but to automate encryption upon detecting contacts, recognizing having the public keys, and the messages autocrypt headers, using the encryption already supported by TB.

**Comment by dadada** [comment]
If anyone was wondering, you can export your keys without the help of the Add-On.

*tl;dr* Get the passphrase from Thunderbird's stored logins and `sqlite3 $HOME/.thunderbird/*/autocrypt.sqlite <<< 'select hex(key_data_secret) from secret_keydata;' | xxd -r -p | gpg --import`.

First [make a backup of your profile](https://support.mozilla.org/en-US/kb/profiles-where-thunderbird-stores-user-data) of your Thunderbird profile. Then you can export your keys using

```
sqlite3 $PATH_TO_YOUR_PROFILE/autocrypt.sqlite <<< 'select hex(key_data_secret) from secret_keydata;' | xxd -r -p  > autocrypt-exported-packets
```

Your may then review the exported packets using `gpg --list-packets autocrypt-exported-packets` and import them into your keyring.

```
gpg --import autocrypt-exported-packets
```

It will prompt for a decryption passphrase which is stored in Thunderbird's saved [saved passwords](https://support.mozilla.org/en-US/questions/1181295#answer-1020603). The provider is *chrome://autocrypt (DO NOT DELETE)* and the user should be *autocrypt*.

Thanks @Valodim

**Comment by saltedlolly** [comment]
I am confused by the above. Are there plans to get the addon working with Thunderbird 78?

**Comment by bam80** [comment]
Ping?

**Comment by Valodim** [comment]
Unfortunately, Thunderbird 78 removed support for extensions that we need to make this work :( So it's technically not possible to update it for TB78.

Instead TB gained first-party support for OpenPGP, which I was hoping would gain Autocrypt support. But at least until now they have very much stuck to the "traditional" PGP user experience, which honestly didn't do much for my motivation to work on the Autocrypt ecosystem.

So sadly, the Autocrypt extension never made it out of "experimental" status. I'll mark it as deprecated. I apologize for not being very communicative about this before. Thanks everyone who used the extension! It was a good run!

\\\\ edit

(actually I'll leave this issue open so it's easier to find)

**Comment by brenthuisman** [comment]
Thanks for the update, and thanks for the plugin, for as long as we could use it!
